src/core/sysroot.rs

```diff
@@ -0,0 +1,17 @@
+use std::{env, path::PathBuf, process::Command, sync::OnceLock};
+
+static SYSROOT: OnceLock<Option<PathBuf>> = OnceLock::new();
+
+pub(crate) fn get_sysroot() -> &'static Option<PathBuf> {
+    SYSROOT.get_or_init(|| {
+        let Ok(rustc) = env::var("RUSTC") else {
+            return None;
+        };
+        Command::new(rustc)
+            .arg("--print=sysroot")
+            .output()
+            .map(|x| String::from_utf8_lossy(&x.stdout).trim().to_owned())
+            .map(PathBuf::from)
+            .ok()
+    })
+}
```

Comment:
Is the sysroot dependent on `--target`?  If so, we'd need to do one pass without it and one with the `--target` we forward to `cargo check`